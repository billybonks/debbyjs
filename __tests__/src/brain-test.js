const describeClass = require('../helpers/class');

describeClass(() => {

  beforeEach(() => {
    subject.userStore.fetch = jest.fn();
    subject.userStore.save = jest.fn();
    subject.contextStore.fetch = jest.fn();
    subject.contextStore.save = jest.fn();
  });
  describe('#getUser', () => {
    test('propergates to correct userStore.fetch', () => {
      subject.getUser(12);
      expect(subject.userStore.fetch.mock.calls.length).toEqual(1);
      expect(subject.userStore.fetch.mock.calls[0]).toMatchSnapshot();
    });
  });

  describe('#saveUser', () => {
    test('propergates to correct userStore.save', () => {
      subject.saveUser(18, {name:'seb', account:1});
      expect(subject.userStore.save.mock.calls.length).toEqual(1);
      expect(subject.userStore.save.mock.calls[0]).toMatchSnapshot();
    });
  });

  describe('#getContext', () => {
    test('propergates to correct contextStore.fetch', () => {
      subject.getContext(22);
      expect(subject.contextStore.fetch.mock.calls.length).toEqual(1);
      expect(subject.contextStore.fetch.mock.calls[0]).toMatchSnapshot();
    });
  });


  describe.only('#saveContext', () => {
    test('propergates to correct contextStore.save', () => {
      subject.saveContext(18, {name:'cooldude', account:99, word:true});
      expect(subject.contextStore.save.mock.calls.length).toEqual(1);
      expect(subject.contextStore.save.mock.calls[0]).toMatchSnapshot();
    });
  });
});
