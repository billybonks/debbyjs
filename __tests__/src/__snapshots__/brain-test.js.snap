// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit | Bot instance functions #handleMessage if finds string uses it 1`] = `
Object {
  "context": Object {
    "lastMessage": null,
  },
  "response": "fallback.assertion",
}
`;

exports[`Unit | Bot instance functions #handleMessage if function executes it 1`] = `"thebest.message"`;

exports[`Unit | Bot instance functions #handleMessage if no message found uses fallback 1`] = `
Object {
  "context": Object {
    "lastMessage": null,
  },
  "response": "fallback",
}
`;
